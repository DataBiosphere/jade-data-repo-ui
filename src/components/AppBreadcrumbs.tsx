import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Breadcrumbs, Link } from '@material-ui/core';
import { Link as RouterLink } from 'react-router-dom';

function capitalize(str: string): string {
  return str.charAt(0).toUpperCase() + str.slice(1);
}

const styles = (theme: any) => ({
  breadcrumb: {
    fontStyle: 'italic',
  },
  terminalBreadcrumb: {
    fontStyle: 'italic',
    color: theme.palette.primary.dark,
    cursor: 'default',
    '&:hover': {
      textDecoration: 'none',
    },
  },
});

type Context = { type: string; id: string; name: string };

type AppBreadcrumbsProps = {
  links: Array<Function>;
  context: Context;
};

const BreadcrumbsLink = withStyles(styles)((props: any) => {
  const { classes, ...other } = props;
  return <Link component={RouterLink} color="primary" className={classes.breadcrumb} {...other} />;
});

const TerminalBreadcrumb = withStyles(styles)((props: any) => {
  const { classes, ...other } = props;
  return <Link color="primary" className={classes.terminalBreadcrumb} {...other} />;
});

const Breadcrumb = withStyles(styles)((props: any) => {
  if (props.disabled) {
    return <TerminalBreadcrumb {...props} />;
  }
  return <BreadcrumbsLink {...props} />;
});

function dashboardLink(key: string, disabled: boolean) {
  return (
    <Breadcrumb key={key} to="/" disabled={disabled}>
      Dashboard
    </Breadcrumb>
  );
}

function contextLink(context: Context) {
  return (key: string, disabled: boolean) => (
    <Breadcrumb key={key} to={`/${context.type}`} disabled={disabled}>
      {capitalize(context.type)}
    </Breadcrumb>
  );
}

function collectionLink(context: Context) {
  return (key: string, disabled: boolean) => (
    <Breadcrumb key={key} to={`/${context.type}/${context.id}`} disabled={disabled}>
      {context.name}
    </Breadcrumb>
  );
}

export function contextChildLink(child: string, text: string) {
  return (key: string, disabled: boolean) => (
    <Breadcrumb key={key} to={child} disabled={disabled}>
      {text}
    </Breadcrumb>
  );
}

const AppBreadcrumbs: React.FC<AppBreadcrumbsProps> = (props) => {
  const { context, links } = props;

  const contextLinks: Array<Function> = [
    dashboardLink,
    contextLink(context),
    collectionLink(context),
  ];
  const toRender: Array<Function> = [...contextLinks, ...(links || [])];

  return (
    <Breadcrumbs aria-label="breadcrumb">
      {toRender.map((linkFunc, i) => linkFunc(i, i === toRender.length - 1))}
    </Breadcrumbs>
  );
};

export default withStyles(styles)(AppBreadcrumbs);
