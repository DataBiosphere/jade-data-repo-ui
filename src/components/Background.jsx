import styled from 'styled-components';

const Background = styled.div`
  background: #000 linear-gradient(to bottom, #fff 0%, #fff 80%) fixed;
  color: #000;
  min-height: 100vh;
  overflow: hidden;
  position: relative;

  &:before {
    background: linear-gradient(to bottom, #000, #fff);
    bottom: 0;
    content: '';
    left: 0;
    opacity: 0.4;
    position: absolute;
    right: 0;
    top: 0;
    transform: rotate(-20deg) scale(1.5) translate(0, 43%);
  }
`;

export default Background;
