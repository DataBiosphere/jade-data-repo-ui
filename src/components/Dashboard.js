import React, { Component } from 'react'
import { div, h } from 'react-hyperscript-helpers'
import * as Nav from '../nav'

import jade from '../jade.png';
import '../App.css';

class HeroDashboard extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={jade} className="App-logo" alt="jade" />
          <p>
            Welcome to the Jade Data Repository!
          </p>
          <a
            className="App-link"
            href="https://github.com/DataBiosphere/jade-data-repo"
            target="_blank"
            rel="noopener noreferrer"
          >
            See the Jade source code
          </a>
        </header>
      </div>
    )
  }
}

const addNavPaths = () => {
  Nav.defRedirect({ regex: /^.{0}$/, makePath: () => 'dashboard' })
  Nav.defPath(
    'dashboard',
    {
      component: props => h(HeroDashboard, props),
      regex: /dashboard/,
      makeProps: () => {},
      makePath: () => 'dashboard'
    }
  )
}

export { HeroDashboard, addNavPaths }
