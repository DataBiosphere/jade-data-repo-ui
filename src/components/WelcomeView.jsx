import React from 'react';
import PropTypes from 'prop-types';
import { GoogleLogin } from 'react-google-login';
import { withStyles } from '@material-ui/core/styles';

import Hero from 'assets/media/images/hero.png';
import { logOut, logIn } from 'actions/index';

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    color: theme.palette.primary.main,
    fontSize: '54px',
    lineHeight: '66px',
    paddingBottom: theme.spacing.unit * 8,
  },
  mainContent: {
    display: 'inline-block',
    color: theme.typography.color,
    overflow: 'hidden',
    padding: theme.spacing.unit * 10,
    width: '60%',
  },
  newUser: {
    color: theme.palette.secondary.contrastText,
    fontSize: '18px',
    fontWeight: '300',
    paddingBottom: theme.spacing.unit * 2,
    paddingTop: theme.spacing.unit * 8,
  },
  terraLink: {
    color: theme.palette.primary.main,
    paddingBottom: theme.spacing.unit * 4,
    paddingTop: theme.spacing.unit * 2,
    textDecoration: 'none',
  },
  terraLink: {
    color: theme.palette.primary.main,
    paddingBottom: theme.spacing.unit * 4,
    paddingTop: theme.spacing.unit * 2,
    textDecoration: 'none',
  },
  header: {
    fontSize: '28px',
    lineHeight: '36px',
  },
  heroContainer: {
    display: 'inline-block',
  },
  hero: {
    width: '500px',
  },
  warning: {
    fontWeight: '900',
    paddingBottom: theme.spacing.unit * 2,
<<<<<<< HEAD
  },
=======
  }
>>>>>>> styling pass on welcome page
});

class WelcomeView extends React.PureComponent {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  render() {
    const onSignInFailure = () => {
      const { dispatch } = this.props;
      dispatch(logOut());
    };

    const onSignInSuccess = () => {
      const { dispatch } = this.props;
      dispatch(logIn());
    };
    const { classes } = this.props;

    return (
      <div className={classes.container}>
        <div className={classes.mainContent}>
          <div className={classes.title}>
          Welcome to the Jade Data Repository
          </div>
          <div>
            <GoogleLogin // stealing clientId from Terra
              clientId="500025638838-s2v23ar3spugtd5t2v1vgfa2sp7ppg0d.apps.googleusercontent.com"
              buttonText="Sign in with Google"
              onSuccess={onSignInSuccess}
              onFailure={onSignInFailure}
              theme="dark"
            />
          </div>
          <div className={classes.newUser}>New User?</div>
          <div className={classes.header}>Jade Data Repository requires a Google account.</div>
          <p>
            Terra uses your Google account. Once you have signed in and completed the user profile
            registration step, you can start using Terra.
          </p>
          <div href="https://app.terra.bio/" className={classes.terraLink}>
            Need to create a Terra account?
          </div>
          <hr />
          <div>
            <div>
              <div className={classes.warning}>WARNING NOTICE</div>
              <div>
                <p>
                  You are accessing a US Government web site which may contain information that must
                  be protected under the US Privacy Act or other sensitive information and is
                  intended Government authorized use only.
                </p>
                <p>
                  Unauthorized attempts to upload information, change information, or use of this
                  web site may result in disciplinary action, civil, and/or criminal penalties.
                  Unauthorized users of this website should have no expectation of privacy regarding
                  any communications or data processed by this website.
                </p>
                <p>
                  Anyone accessing this website expressly consents to monitoring of their actions
                  and all communications or data transiting or stored on related to this website and
                  is advised that if such monitoring reveals possible evidence of criminal activity,
                  NIH may provide that evidence to law enforcement officials.
                </p>
                <p>WARNING NOTICE (when accessing TCGA controlled data)</p>
                <p>
                  You are reminded that when accessing TCGA controlled accessinformation you are
                  bound by the dbGaP TCGA DATA USE CERTIFICATION AGREEMENT (DUCA).
                </p>
              </div>
            </div>
          </div>
        </div>

        <div className={classes.heroContainer}>
          <img src={Hero} alt="hero" className={classes.hero} />
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(WelcomeView);
