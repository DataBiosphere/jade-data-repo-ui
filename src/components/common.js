import _ from 'lodash/fp'
import marked from 'marked'
import { Fragment } from 'react'
import { div, h, input, label, span } from 'react-hyperscript-helpers'
import Interactive from 'react-interactive'
import RSelect from 'react-select'
import { centeredSpinner, icon } from './icons'
import TooltipTrigger from './TooltipTrigger'
import colors from 'src/libs/colors'
import { getConfig } from 'src/libs/config'
import * as Style from 'src/libs/style'


const styles = {
  button: {
    display: 'inline-flex', justifyContent: 'space-around', alignItems: 'center', height: '2.25rem',
    fontWeight: 500, fontSize: 14, textTransform: 'uppercase', whiteSpace: 'nowrap',
    userSelect: 'none'
  }
}

export const Clickable = ({ as = 'div', disabled, tooltip, tooltipSide, onClick, children, ...props }) => {
  const child = h(Interactive, _.merge({
    as, disabled,
    onClick: (...args) => onClick && !disabled && onClick(...args)
  }, props), [children])

  if (tooltip) {
    return h(TooltipTrigger, { content: tooltip, side: tooltipSide }, [child])
  } else {
    return child
  }
}

const linkProps = disabled => ({
  as: 'a',
  style: {
    color: disabled ? colors.gray[2] : colors.blue[0],
    cursor: disabled ? 'not-allowed' : 'pointer'
  },
  hover: disabled ? undefined : { color: colors.blue[1] }
})

export const link = function(props, children) {
  return h(Interactive,
    _.merge(linkProps(props.disabled), props),
    children)
}

export const linkButton = (props, children) => {
  return h(Clickable,
    _.merge(linkProps(props.disabled), props),
    children)
}

export const buttonPrimary = ({ disabled, ...props }, children) => {
  return h(Clickable, _.merge({
    disabled,
    style: {
      ...styles.button,
      borderRadius: 5, color: 'white', padding: '0 1.5rem',
      fontWeight: 'bold',
      backgroundColor: disabled ? colors.gray[2] : colors.blue[0],
      cursor: disabled ? 'not-allowed' : 'pointer'
    },
    hover: disabled ? undefined : { backgroundColor: colors.blue[1] }
  }, props), children)
}

export const buttonSecondary = ({ disabled, ...props }, children) => {
  return h(Clickable, _.merge({
    disabled,
    style: {
      ...styles.button,
      color: disabled ? colors.gray[2] : colors.gray[0],
      fontWeight: 'bold',
      cursor: disabled ? 'not-allowed' : 'pointer'
    },
    hover: disabled ? undefined : { color: colors.gray[1] }
  }, props), children)
}

export const search = function({ wrapperProps, inputProps }) {
  return div(
    _.merge({ style: { borderBottom: '1px solid black', padding: '0.5rem 0', display: 'flex' } },
      wrapperProps),
    [
      icon('search', { size: 21 }),
      input(_.merge({
        style: {
          border: 'none', outline: 'none',
          flexGrow: 1,
          verticalAlign: 'bottom', marginLeft: '1rem',
          fontSize: '1rem'
        }
      }, inputProps))
    ])
}

export const tabBar = ({ activeTab, tabNames, refresh = _.noop, getHref }, children = []) => {
  const navSeparator = div({
    style: { background: 'rgba(255,255,255,0.15)', width: 1, height: '3rem', flexShrink: 0 }
  })

  const navTab = currentTab => {
    const selected = currentTab === activeTab
    const href = getHref(currentTab)
    const hideSeparator = selected || tabNames.indexOf(activeTab) === tabNames.indexOf(currentTab) + 1

    return h(Fragment, [
      h(Interactive, {
        as: 'a',
        style: { ...Style.tabBar.tab, ...(selected ? Style.tabBar.active : {}) },
        hover: { color: Style.tabBar.active.color },
        onClick: href === window.location.hash ? refresh : undefined,
        href
      }, currentTab),
      !hideSeparator && navSeparator
    ])
  }

  return div({ style: Style.tabBar.container }, [
    activeTab !== tabNames[0] && navSeparator,
    ..._.map(name => navTab(name), tabNames),
    div({ style: { flexGrow: 1 } }),
    ...children
  ])
}

export const menuIcon = (iconName, props) => {
  return icon(iconName, _.merge({ size: 15, style: { marginRight: '.5rem' } }, props))
}

export const MenuButton = ({ disabled, children, ...props }) => {
  return h(Clickable, _.merge({
    disabled,
    style: {
      display: 'flex', alignItems: 'center',
      fontSize: 12, minWidth: 125, height: '2rem',
      color: disabled ? colors.gray[2] : undefined,
      padding: '0 1.5rem',
      cursor: disabled ? 'not-allowed' : 'pointer'
    },
    hover: !disabled ? { backgroundColor: colors.blue[3], fontWeight: 'bold' } : undefined
  }, props), [children])
}

export const Checkbox = ({ checked, onChange, disabled, ...props }) => {
  return h(Interactive, {
    as: 'span',
    role: 'checkbox',
    'aria-checked': checked,
    onClick: () => onChange && !disabled && onChange(!checked),
    style: {
      display: 'inline-flex',
      verticalAlign: 'middle',
      color: disabled ? colors.gray[2] : colors.blue[0]
    },
    hover: disabled ? undefined : { color: colors.blue[1] },
    active: disabled ? undefined : { backgroundColor: colors.blue[5] },
    disabled,
    ...props
  }, [
    icon(checked ? 'checkSquare' : 'square', { size: 16 })
  ])
}

export const LabeledCheckbox = ({ checked, onChange, disabled, children, ...props }) => {
  return h(Fragment, [
    h(Checkbox, { checked, onChange, disabled, ...props }),
    h(Interactive, {
      as: 'span',
      style: { verticalAlign: 'middle' },
      onClick: () => onChange && !disabled && onChange(!checked)
    }, [children])
  ])
}

export const RadioButton = ({ text, labelStyle, ...props }) => {
  const id = `${text}-radio-button`

  return h(Fragment, [
    input({
      type: 'radio', id,
      name: id, // not semantically correct, but fixes a focus cycle issue
      ...props
    }),
    label({ htmlFor: id, style: labelStyle }, text)
  ])
}

export const spinnerDefault = ({ outerStyles = {}, innerStyles = {} }) => div(
  {
    style: {
      position: 'absolute',
      display: 'flex', alignItems: 'center',
      top: 0, right: 0, bottom: 0, left: 0,
      zIndex: 9999, // make sure it's on top of any third party components with z-indicies
      ...outerStyles
    }
  }, [
    centeredSpinner({
      size: 64,
      style: { backgroundColor: 'rgba(255, 255, 255, 0.85)', padding: '1rem', borderRadius: '0.5rem', ...innerStyles }
    })
  ]
)

export const spinnerOverlay = spinnerDefault({ outerStyles: { backgroundColor: 'rgba(0, 0, 0, 0.1)' } })

export const transparentSpinnerOverlay = spinnerDefault({ innerStyles: { backgroundColor: 'rgba(255, 255, 255, 0.0)' } })

export const topSpinnerOverlay = spinnerDefault({ outerStyles: { backgroundColor: 'rgba(0, 0, 0, 0.1)' }, innerStyles: { marginTop: 150  } })

export const comingSoon = span({
  style: {
    margin: '0.5rem', padding: 3, borderRadius: 2,
    backgroundColor: colors.purple[0], color: 'white',
    fontSize: '75%', textTransform: 'uppercase', fontWeight: 500,
    whiteSpace: 'nowrap', lineHeight: 1
  }
}, ['coming soon'])

/**
 * @param {Object} props - see {@link https://react-select.com/props#select-props}
 * @param props.value - a member of options
 * @param {Array} props.options - can be of any type; if objects, they should each contain a value and label, unless defining getOptionLabel
 */
export const Select = ({ value, options, ...props }) => {
  const newOptions = options && !_.isObject(options[0]) ? _.map(value => ({ value }), options) : options
  const findValue = target => _.find({ value: target }, newOptions)
  const newValue = props.isMulti ? _.map(findValue, value) : findValue(value)

  return h(RSelect, _.merge({
    theme: base => _.merge(base, {
      colors: {
        primary: colors.blue[0],
        neutral20: colors.gray[3],
        neutral30: colors.gray[3]
      },
      spacing: { controlHeight: 36 }
    }),
    styles: {
      control: (base, { isDisabled }) => _.merge(base, {
        backgroundColor: isDisabled ? colors.gray[5] : 'white',
        boxShadow: 'none'
      }),
      singleValue: base => ({ ...base, color: colors.gray[0] }),
      option: (base, { isSelected, isFocused, isDisabled }) => _.merge(base, {
        backgroundColor: isSelected ? colors.blue[4] : isFocused ? colors.blue[5] : undefined,
        fontWeight: isFocused ? 'bold' : undefined,
        color: isDisabled ? undefined : colors.gray[0],
        ':active': { backgroundColor: isSelected ? colors.blue[4] : colors.blue[5] }
      }),
      clearIndicator: base => ({ ...base, paddingRight: 0 }),
      indicatorSeparator: () => ({ display: 'none' }),
      dropdownIndicator: base => _.merge(base, { paddingLeft: props.isClearable ? 0 : undefined }),
      multiValueRemove: base => _.merge(base, { ':hover': { backgroundColor: 'unset' } })
    },
    getOptionLabel: ({ value, label }) => label || value.toString(),
    value: newValue,
    options: newOptions
  }, props))
}

export const FadeBox = ({ fadePoint = '60%', style = {}, children }) => {
  const {
    paddingTop = '1.5rem',
    paddingLR = '1.5rem',
    borderRadius = '8px',
    backgroundColor = 'rgba(255,255,255,0)',
    borderColor = colors.gray[3],
    ...containerStyle
  } = style

  return div({
    style: {
      display: 'flex', flexDirection: 'column',
      background: `linear-gradient(to bottom, white, ${backgroundColor} ${fadePoint})`,
      borderRadius: `${borderRadius} ${borderRadius} 0 0`,
      ...containerStyle
    }
  }, [
    div({
      style: {
        height: paddingTop,
        border: `1px solid ${borderColor}`,
        borderBottom: 'none',
        borderRadius: `${borderRadius} ${borderRadius} 0 0`
      }
    }),
    div({
      style: {
        flex: 1,
        padding: `0 ${paddingLR}`,
        borderWidth: 1,
        borderStyle: 'solid',
        borderImage: `linear-gradient(to bottom, ${borderColor}, ${backgroundColor} ${fadePoint}) 1 100%`,
        borderTop: 'none',
        borderBottom: 'none'
      }
    }, [children])
  ])
}

export const PageFadeBox = ({ children, style = {} }) => {
  return h(FadeBox, {
    fadePoint: '125px',
    style: {
      margin: '1.5rem', paddingTop: '1rem', minHeight: 125, ...style
    }
  }, [children])
}

export const backgroundLogo = icon('logoIcon', {
  size: 1200,
  style: { position: 'fixed', top: -100, left: -100, zIndex: -1, opacity: 0.65 }
})

export const methodLink = config => {
  const { methodRepoMethod: { sourceRepo, methodVersion, methodNamespace, methodName, methodPath } } = config
  return sourceRepo === 'agora' ?
    `${getConfig().firecloudUrlRoot}/#methods/${methodNamespace}/${methodName}/${methodVersion}` :
    `${getConfig().dockstoreUrlRoot}/workflows/${methodPath}`
}

/**
 * WARNING: Be very careful when using custom renderers because they may override marked's built-in
 * content sanitization.
 * @param {string} children markdown content
 * @param renderers element-specific renderers
 * @param props properties for wraper div
 * @returns {object} div containing rendered markdown
 * @constructor
 */
export const Markdown = ({ children, renderers = {}, ...props }) => {
  const content = marked(children, {
    renderer: Object.assign(new marked.Renderer(), renderers)
  })
  return div({ ...props, dangerouslySetInnerHTML: { __html: content } })
}
