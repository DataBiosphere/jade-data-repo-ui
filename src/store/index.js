import { applyMiddleware, createStore, compose, combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import rootSaga from 'sagas/index';
import rootReducer from 'reducers/index';
import history from 'modules/hist';
import middleware, { sagaMiddleware } from './middleware';

const reducer = (hist) =>
  combineReducers({
    router: connectRouter(hist),
    ...rootReducer,
  });

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState = {}) => {
  const store = createStore(
    reducer(history),
    initialState,
    composeEnhancer(applyMiddleware(...middleware)),
  );

  sagaMiddleware.run(rootSaga);

  if (module.hot) {
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return {
    store,
  };
};

const { store } = configStore();

global.store = store;

export { store };
