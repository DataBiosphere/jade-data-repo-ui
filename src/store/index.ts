import { applyMiddleware, combineReducers, compose, createStore, Store } from 'redux';
import { connectRouter } from 'connected-react-router';

import rootSaga from 'sagas/index';
import rootReducer, { initialTdrState, TdrState } from 'reducers/index';
import history from 'modules/hist';
import { History } from 'history';
import middleware, { sagaMiddleware } from './middleware';

const reducer = (hist: History) =>
  combineReducers({
    router: connectRouter(hist),
    ...rootReducer,
  });

// @ts-ignore
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState: TdrState = initialTdrState): Store<TdrState> => {
  const store = createStore(
    reducer(history),
    initialState,
    composeEnhancer(applyMiddleware(...middleware)),
  );

  sagaMiddleware.run(rootSaga);

  // @ts-ignore
  if (module.hot) {
    // @ts-ignore
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return store;
};

const store = configStore();

// @ts-ignore
global.store = store;

export { store };
