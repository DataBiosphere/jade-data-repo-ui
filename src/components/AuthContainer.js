import { h } from 'react-hyperscript-helpers'
import { centeredSpinner } from 'src/components/icons'
import { authStore } from 'src/libs/auth'
import * as Utils from 'src/libs/utils'
import { Disabled, Unlisted } from 'src/pages/Disabled'
import Register from 'src/pages/Register'
import SignIn from 'src/pages/SignIn'
import TermsOfService from 'src/pages/TermsOfService'


export default Utils.connectAtom(authStore, 'authState')(
  ({ children, authState: { isSignedIn, registrationStatus, acceptedTos } }) => {
    return Utils.cond(
      [isSignedIn === undefined, centeredSpinner],
      [isSignedIn === false, h(SignIn)],
      [registrationStatus === undefined, centeredSpinner],
      [registrationStatus === 'unregistered', h(Register)],
      [registrationStatus === 'disabled', () => h(Disabled)],
      [registrationStatus === 'unlisted', () => h(Unlisted)],
      [acceptedTos === undefined, centeredSpinner],
      [acceptedTos === false, () => h(TermsOfService)],
      children
    )
  }
)
