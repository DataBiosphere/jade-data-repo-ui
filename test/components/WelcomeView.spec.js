import React from 'react';
import { Provider } from 'react-redux';

import { store } from 'store/index';
import { WelcomeView } from 'components/WelcomeView';

const mockDispatch = jest.fn();
const ownProps = {
  dispatch: mockDispatch,
  classes: {},
  configuration: {
    clientId: '5',
  },
  users: {},
};

jest.mock('react-google-login/dist/google-login', () => ({
  GoogleLogin: props => {
    const { onSuccess } = props;
    return <button type="button" onClick={onSuccess} />;
  },
}));

function setup() {
  return mount(
    <Provider store={store}>
      <WelcomeView {...ownProps} />
    </Provider>,
  );
}

describe('WelcomeView', () => {
  const wrapper = setup();

  it('should handle login success', () => {
    const mockGoogleLogin = wrapper
      .find('GoogleLogin')
      .find('button')
      .simulate('click');

    expect(mockGoogleLogin).not.toBeNull();
    expect(mockGoogleLogin.length).toEqual(1);
    // expect(mockDispatch).toHaveBeenCalledWith({ payload: {}, type: 'USER_LOGIN' });
    expect(mockDispatch).toHaveBeenCalled();
  });
});
