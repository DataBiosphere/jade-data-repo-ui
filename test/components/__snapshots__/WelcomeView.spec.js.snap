// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeView should render properly 1`] = `
<div>
  <h1>
    Welcome to the Jade Data Repository
  </h1>
  <div>
    <button
      onClick={[Function]}
      type="button"
    />
  </div>
  <h3>
    New User?
  </h3>
  <h3>
    Data Repository Requires a Terra Google account.
  </h3>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo. Proin sodales pulvinar sic tempor.
  </p>
  <a>
    Need to create a Terra account?
  </a>
  <hr />
  <div>
    <div>
      <div>
        Warning Notice
      </div>
      <div>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo. Proin sodales pulvinar sic tempor. Sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nam fermentum, nulla luctus pharetra vulputate, felis tellus mollis orci, sed rhoncus pronin sapien nunc accuan eget.
      </div>
    </div>
  </div>
</div>
`;
